src/qmapshack/poi/CPoiDraw.cpp

```diff
@@ -155,18 +155,14 @@ poi_t CPoiDraw::findPOICloseBy(const QPoint &px) const
                 break;
             }
 
-            if(item->findPoiCloseBy(px, poi))
-            {
-                // stop at the 1st one found
-                break;
-            }
+            item->findPOICloseBy(px, pois, posPOIHighlight);
         }
         CPoiItem::mutexActivePois.unlock();
     }
-    return poi;
+    return pois;
 }
 
-void CPoiDraw::findPoisIn(const QRectF &degRect, QList<poi_t> & pois) const
+void CPoiDraw::findPoisIn(const QRectF &degRect, QList<poi_t> & pois, QList<QPointF>&posPOIHighlight) const
```

Comment:
Pois vs POIs. Please make sure you are using it the same way in all your code. It really helps a lot to stick to constant naming schemes.